src/pyk/kcfg/explore.py

```diff
@@ -130,15 +130,11 @@ def cterm_get_model(self, cterm: CTerm, module_name: str | None = None) -> Subst
             if not result.model:
                 return Subst({})
             model_subst = self.kprint.kore_to_kast(result.model)
-            _subst: dict[str, KInner] = {}
-            for subst_pred in flatten_label('#And', model_subst):
-                subst_pattern = mlEquals(KVariable('###VAR'), KVariable('###TERM'))
-                m = subst_pattern.match(subst_pred)
-                if m is not None and type(m['###VAR']) is KVariable:
-                    _subst[m['###VAR'].name] = m['###TERM']
-                else:
-                    raise AssertionError(f'Received a non-substitution from get-model endpoint: {subst_pred}')
-            return Subst(_subst)
+            try:
+                return Subst.from_pred(model_subst)
+            except ValueError as err:
+                raise AssertionError(f'Received a non-substitution from get-model endpoint: {model_subst}') from err
```

Comment:
Do we need to catch and re-raise this exception?

```suggestion
            return Subst.from_pred(model_subst)
```

The stack trace + original exception should be enough information I guess?

Reply:
The difference is subtle, but the `AssertionError` suggests that there is a bug in the `get-model` endpoint.